### PR TITLE
Fixed inconsistencies regarding ReduceLROnPlateau

### PR DESCRIPTION
**Pull request by sotlampr**
Regarding #9541, and to get your attention :)

### CODE REVIEW

**Comment by Dref360** [comment]
This is a breaking API change. Please ensure backward compatibility. 
Also, please add a test that validates your fix at line 970.

**Comment by fchollet** [comment]
That's a reasonable change, but as stated, we should ensure full backwards compatibility.

File: keras/callbacks.py [hunk 1]
```diff
@@ -898,24 +898,29 @@ class ReduceLROnPlateau(Callback):
...
             to only focus on significant changes.
         cooldown: number of epochs to wait before resuming
             normal operation after lr has been reduced.
         min_lr: lower bound on the learning rate.
     """
 
     def __init__(self, monitor='val_loss', factor=0.1, patience=10,
-                 verbose=0, mode='auto', epsilon=1e-4, cooldown=0, min_lr=0):
+                 verbose=0, mode='auto', min_delta=1e-4, cooldown=0, min_lr=0,
+                 epsilon=None):
```
Comment (Dref360) [inline keras/callbacks.py]: use **kwargs to be backward compatible

File: tests/keras/test_callbacks.py [hunk 2]
```diff
@@ -334,18 +334,43 @@ def make_model():
...
+def test_ReduceLROnPlateau_patience():
+    class DummyOptimizer(object):
+        def __init__(self):
+            self.lr = K.variable(1.0)
+
+    class DummyModel(object):
+        def __init__(self):
+            self.optimizer = DummyOptimizer()
+
+    reduce_on_plateau = callbacks.ReduceLROnPlateau(monitor='val_loss',
```
Comment (Dref360) [inline tests/keras/test_callbacks.py]: Last thing :P 
Could you add a test which validate the backward compatibility? Should have said this earlier.

Comment (sotlampr) [reply tests/keras/test_callbacks.py]: Do you also test for deprecation warnings?

Comment (Dref360) [reply tests/keras/test_callbacks.py]: Good catch! yeah there should be a warning.

**Pull request merged**